docs/blog/2023-07-15-incremental.md

```diff
@@ -0,0 +1,235 @@
+possible.  Also, unlike Word, `diff` does not show character changes,
+only full line changes.
+
+Also, notice that `diff` changes are described as being composed of
+multiple simpler changes: the example contains one insertion, one
+deletion, and one modification.
+
+`diff` can compare not only two files but it can also compare multiple
+files at once.  Entire files can be inserted, deleted, renamed, or
+changed individually.  Another tool,
+[patch](https://en.wikipedia.org/wiki/Patch_(Unix)) can use such a
+description to convert a set of old file versions into a set of newer
+ones.
+
+> Changes can be hierarchical: changes to complex objects can be
+  described from changes to simpler ones
+
+## Data and changes
+
+Notice that we are dealing with two different kinds of structures: the
+changed objects (e.g., files), and the changes between objects
+(deltas, or diffs).  (One can store a change in a text file, but it's
+really a different object.)
+
+Let's call these two data structures $D$ for data, and $\Delta$ for
+changes.  First we have a *diff* operation $D - D \to \Delta$: where
+we use minus for the *diff*.  The diff between two data values is a
+change.
+
+We also have an inverse operation: $D + \Delta \to D$, which changes
+some data by *applying* a change.  There is also a notion of an *empty
+change*, which is the difference between two identical data values.
+
+## Changing numbers
+
+In some circumstances the data and the changes can look similar.
+Consider the case of a bank account.  The value of the account is a
+dollar amount -- a number.  At the end of each month we get a bank
+statement describing the change in the value of the account: also a
+number.  For bank accounts the changes can be both positive and
+negative, but, unless you have overdraft protection, the amounts are
+always positive.
+
+This is another important lesson: applying a change to a legal data
+value may not produce another legal data value.  Removing $10 from
+your bank account is a legal change if you have $20, but not if you
+have only $5 left.  So the validity of a change cannot be evaluated in
+isolation, you have to know *what* is being changed.  Only the
+application domain can decide that.
+
+For this case the structures $D$ and $\Delta$ are identical: you can
+represent both the data and the changes as numbers (but you must
+include negative numbers if you want to represent all possible
+changes).  *Empty* data and *empty* changes are both the number zero.
+This is very convenient.  We will show up in a future blog post that
+we can take advantage of this nice property even when we don't handle
+just numbers.
+
+## Databases and changes
+
+Databases are an application domain where changes are actually
+central.  Changes appear in databases in two guises, which look very
+dissimilar superficially, which we describe below.  We think that this
+is a historical mistake, and the two notions should be unified.
```

Comment:
I thought that it would be good to be explicit about what these two guises are in advance, because otherwise I want to skip ahead to find out and nothing otherwise explicitly says (I guess the sectioning structure sort-of does).

The rest of my suggested change is just wordsmithing. Feel free to drop that if you don't like my wording.

```suggestion
central.  Changes appear in databases in two guises, those of
transactions and change data capture, which we describe in
detail below.  These forms appear superficially dissimilar and
historically treated separately, but their fundamental 
similarity leads us to believe that in the future they should be
unified.
```